mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageModule.java

```diff
@@ -0,0 +1,122 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres.mail;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import org.apache.james.backends.postgres.PostgresIndex;
+import org.apache.james.backends.postgres.PostgresModule;
+import org.apache.james.backends.postgres.PostgresTable;
+import org.apache.james.mailbox.postgres.mail.PostgresCommons.DataTypes;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.jooq.postgres.extensions.types.Hstore;
+
+public interface PostgresMessageModule {
+
+    Field<UUID> MESSAGE_ID = DSL.field("message_id", SQLDataType.UUID.notNull());
+
+    interface MessageTable {
+        Table<Record> TABLE_NAME = DSL.table("message");
+        Field<UUID> MESSAGE_ID = PostgresMessageModule.MESSAGE_ID;
+        Field<String> BLOB_ID = DSL.field("blob_id", SQLDataType.VARCHAR(200).notNull());
+        Field<String> MIME_TYPE = DSL.field("mime_type", SQLDataType.VARCHAR(200));
+        Field<String> MIME_SUBTYPE = DSL.field("mime_subtype", SQLDataType.VARCHAR(200));
+        Field<LocalDateTime> INTERNAL_DATE = DSL.field("internal_date", DataTypes.TIMESTAMP);
+        Field<Long> SIZE = DSL.field("size", SQLDataType.BIGINT.notNull());
+        Field<Integer> BODY_START_OCTET = DSL.field("body_start_octet", SQLDataType.INTEGER.notNull());
+        Field<byte[]> HEADER_CONTENT = DSL.field("header_content", SQLDataType.BLOB.notNull());
+        Field<Integer> TEXTUAL_LINE_COUNT = DSL.field("textual_line_count", SQLDataType.INTEGER);
+        Field<Hstore> CONTENT_TYPE_PARAMETERS = DSL.field("content_type_parameters", DataTypes.HSTORE);
+        Field<Hstore> CONTENT_DISPOSITION_PARAMETERS = DSL.field("content_disposition_parameters", DataTypes.HSTORE);
+
+        PostgresTable TABLE = PostgresTable.name(TABLE_NAME.getName())
+            .createTableStep(((dsl, tableName) -> dsl.createTableIfNotExists(tableName)
+                .column(MESSAGE_ID)
+                .column(BLOB_ID)
+                .column(MIME_TYPE)
+                .column(MIME_SUBTYPE)
+                .column(INTERNAL_DATE)
+                .column(SIZE)
+                .column(BODY_START_OCTET)
+                .column(HEADER_CONTENT)
+                .column(TEXTUAL_LINE_COUNT)
+                .column(CONTENT_TYPE_PARAMETERS)
+                .column(CONTENT_DISPOSITION_PARAMETERS)
+                .constraint(DSL.primaryKey(MESSAGE_ID))
+                .comment("Holds the metadata of a mail")))
+            .supportsRowLevelSecurity();
+    }
+
+    interface MessageToMailboxTable {
```

Comment:
We need to add the `SIZE` metadata into this table. Then with `RETURNING` keyword we can build the `MessageMetaData` of deletes and be consistent without transactions.